crates/wasmtime/src/runtime/type_registry.rs

```diff
@@ -806,12 +809,12 @@ impl TypeRegistryInner {
             "type is not canonicalized for runtime usage: {ty:?}"
         );
 
-        let gc_layout = match &ty.composite_type {
```

Comment:
This I think we concluded might be able to be replaced with the trait helper method above.